galsim/bandpass.py

```diff
@@ -211,12 +212,13 @@ def _initialize_tp(self):
                     test_wave = 700
                 try:
                     self._tp = galsim.utilities.math_eval('lambda wave : ' + self._orig_tp)
-                    self._tp(test_wave)
+                    from numbers import Real
+                    assert isinstance(self._tp(test_wave), Real)
```

Comment:
Sure.  Thanks for the clarification.  I've switched this `assert`, and the one in sed.py to `isinstance` checks.